segitiga/detriangle.js

```diff
@@ -0,0 +1,167 @@
+var express = require("express");
+var router = express();
+var bodyParser = require("body-parser");
+var getIP = require("ipware")().get_ip;
+var conf = require("./config.json");
+var request = require("superagent");
+
+/* Body Parser */
+router.use(bodyParser.json());
+router.use(bodyParser.urlencoded({
+  extended: true
+}));
+
+/* Inquiry Credit */
+router.post("/", function (req, res, next) {
+  /* Get IP first */
+  var ipInfo = getIP(req);
+  var clientip = ipInfo.clientIp;
+  /* Request Body */
+  var dataReq = {
+    sisi1: req.body.sisi1,
+    sisi2: req.body.sisi2,
+    sisi3: req.body.sisi3,
+  };
+  console.log(
+    "Request to API: " + conf.url.inqA + " " + JSON.stringify(dataReq)
+  );   
+
+  // variabel regex untuk mengecek abjad didalam string dan alfabet didalam string (sesuai kondisi)
+  const regexfornum = /^[0-9]*$/; //untuk mengecek angka
+  const regexforalp = /[a-zA-Z\\s]/g; //untuk mengecek huruf besar dan kecil
+  const regex_symbols = /[-!$%^&*()_+|~=`{}\[\]:\/;<>?,.@#]/; //untuk mengecek character
+  const regex_char = /^[a-zA-Z0-9!@#\$%\^\&*\)\(+=._-]+$/g; //untuk mengecek character,angka, dan abjad
```

Comment:
kenapa tidak pakai yang ini langsung untuk perkondisiannya ? jadi tidak panjang buat perkondisiannya dibawah, langsung ngecek semuanya karakter alphabet dan karakter simbol

Reply:
dan juga spasi

Reply:
di regexforalp dia udah handling spasi, jadi gak perlu buat regex for spasi lagi

Reply:
> kenapa tidak pakai yang ini langsung untuk perkondisiannya ? jadi tidak panjang buat perkondisiannya dibawah, langsung ngecek semuanya karakter alphabet dan karakter simbol

ada beberapa variabel diatas yang kupake dan juga gakupake karena waktu kucoba-coba ada regex yang gabisa kugabung chi, mungkin kalo ada cara bisa gabunginnya regex tsb aku boleh diajarkan chi

Reply:
ohya sama aku masih cari cara buat cek kondisi dimana kalo isi requestnya bukan tipe data string

Reply:
var regForNum = new RegExp('^[0-9]+$')
var regForAlpha = new RegExp('^[a-zA-Z\\s]+$')

pakai ini coba yo

Reply:
okee yo kalo bisanya gitu